src/streamingCalls/streaming.ts

```diff
@@ -113,12 +121,29 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
    * @param {Stream} stream - The API request stream.
    */
   forwardEvents(stream: Stream) {
+    // Avoid "response" events emit twice.
+    let responseHasSent = false;
+
     const eventsToForward = ['metadata', 'response', 'status'];
 
     eventsToForward.forEach(event => {
       stream.on(event, this.emit.bind(this, event));
     });
 
+    // gRPC guaranteed emit the 'status' event but not 'metadata'. Emit the 'response'
+    // event if stream has no 'metadata' event. This also avoid stream swallow the
+    // other events, such as 'end'.
+    stream.on('status', (status: Status) => {
+      if (!responseHasSent) {
+        stream.emit('response', {
+          code: 200,
+          details: '',
+          message: 'OK',
+          metadata: status.metadata,
```

Comment:
@bcoe @alexander-fenster @JustinBeckwith Can I get the feedback on the object of `response` event. 
What I am doing here is try to consistent with response object see L156-L160, but seems like the `status` received here has same structure, but different code. For example, `status` receive using local Java emulator
```
{
  code: 0,
  details: '',
  metadata: Metadata {
    internalRepr: Map(2) { 'content-type' => [Array], 'trailer' => [Array] },
    options: {}
  }
}
```
In my opinion, we should return status as it is. Change status code to `200` cause confusion. But both `status` and `response` event emit same data. So I am not too sure from the user perspectively.

Any suggestion? Thank you .

Reply:
I think the code here is the grpc code not http:
https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
Actually, nvm, I think I'm wrong:
https://screenshot.googleplex.com/GXJfqu8DKHq3DFF

Reply:
I would recommend not including `status.metadata` here. It contains different information than what you would get from the `metadata` event. I would also recommend otherwise matching the `response` event emitted in the `metadata` handler, instead of trying to use the information from the `status`. I think it is not really correct to say that "both `status` and `response` event emit same data". They have similar structures, and the specific fields can match in some cases, but they're not really describing the same thing: `response` seems to describe whether the response *started* successfully, while `status` describes whether the response *completed* successfully.

And just to make sure it is clear, the `code` in the `status` event is a gRPC status code, and the code in the `response` event is an HTTP status code. The HTTP status code is sent in the gRPC protocol, but it is not exposed in the gRPC API.

Reply:
It seems like remapping `0` -> `200` is reasonable, but should we support other grpc status codes and check the repsonse?

Would look to @murgatroid99's advice.

Reply:
Try to do the same thing in [code](https://github.com/googleapis/gax-nodejs/blame/main/src/streamingCalls/streaming.ts#L131-L136), with removing the metadata.
```
stream.emit('response', {
        code: 200,
        details: '',
        message: 'OK',
      });
```